eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -130,17 +131,25 @@ object Graph {
           current = prev.get(current.desc.a)
         }
 
-        edgePath.reverse
+        edgePath
       }
     }
   }
 
-  private def edgeWeightByAmount(edge: GraphEdge, amountMsat: Long): Long = {
-    nodeFee(edge.update.feeBaseMsat, edge.update.feeProportionalMillionths, amountMsat)
+  /**
+    *
+    * @param edge the edge for which we want to compute the weight
+    * @param amountWithFees the value that this edge will have to carry along
+    * @param isNeighborTarget true if the receiving vertex of this edge is the target node (source in a reversed graph), which has cost 0
+    * @return the new amount updated with the necessary fees for this edge
+    */
+  private def edgeWeight(edge: GraphEdge, amountWithFees: Long, isNeighborTarget: Boolean): Long = isNeighborTarget match {
```

Comment:
would an `if` be more optimal here?

Reply:
Depends on which one you consider more readable, i've used 'match' for testing conditions and this is consistent to it.